core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -192,18 +196,27 @@ private ArtifactSignerProvider loadSigners(
               }
 
               slashingProtection.ifPresent(
-                  slashingProtection -> {
-                    slashingProtection.registerValidators(validators);
-                    if (pruningEnabled) {
-                      slashingProtection.prune();
-                    }
-                  });
+                  slashingProtection -> slashingProtection.registerValidators(validators));
               return signers;
             });
 
     return artifactSignerProvider;
   }
 
+  @Override
+  public void run() {
+    super.run();
+    if (pruningEnabled && slashingProtection.isPresent()) {
+      final DbPrunerRunner dbPrunerRunner =
+          new DbPrunerRunner(
+              slashingProtectionParameters,
+              slashingProtection.get(),
+              Executors.newScheduledThreadPool(1));
+      dbPrunerRunner.execute();
+      dbPrunerRunner.schedule();
+    }
+  }
+
```

Comment:
we may extract out this logic in a separate method, say `scheduleAndExecuteDbPruning` or something similar to keep the run method more readable.

Reply:
done